tcping.go

```diff
@@ -114,6 +114,7 @@ func processUserInput(tcpStats *stats) {
 	retryHostnameResolveAfter := flag.Uint("r", 0, "retry resolving target's hostname after <n> number of failed requests. e.g. -r 10 for 10 failed probes.")
 	shouldCheckUpdates := flag.Bool("u", false, "check for updates.")
 	outputJson := flag.Bool("j", false, "output in JSON format.")
+	prettyJson := flag.Bool("p", false, "use pretty json output with indentation.")
```

Comment:
Nothing. When it passed it just calls `jsonEncoder.SetIndent`, which is used in json output only. Although indent for the encoder will change, since `-j` is not passed, it just won't be used for output.

Maybe it's a good idea to add an extra flag check in `if *prettyJson {`

Reply:
Ok, I have two suggestions:

1. Replace the `-p` to `--pretty` to make it a bit less confusing for less tech-savy users (they might confuse it for port number).
2. If `--pretty` is passed without `-j`, call `usage()` and exit (which is implicit when calling `usage`).

Reply:
Will do

Reply:
@pouriyajamshidi Do you want me to keep the `-p` flag and make it available by using either `-p` or `--pretty`, or just `--pretty`?

Personally, I would suggest keeping the `-p` flag primarily because **all** the other flags are one character long. Making this specific flag only support the long version would probably seem weird.

Reply:
We are very likely to make `-p` available for specifying port numbers in the future. Then it will become inconvenient for the current users.

My preference is to make `--pretty` a sub-flag of `-j`. sub-flags are usually not a single character and makes it easier to remember.

Reply:
Makes sense. Will do that way then